### PR TITLE
Feature/make comp v2

### PR DESCRIPTION
**Pull request by srastatter**
v1.1.0: Major rework to the component and pipeline interface definition. Please view Changelog for specific details

### CODE REVIEW

File: AutoMLOps/BuilderUtils.py [hunk 1]
```diff
@@ -329,5 +252,19 @@ def update_params(params: list) -> list:
             param['type'] = python_kfp_types_mapper[param['type']]
         except KeyError as err:
             raise ValueError(f'Unsupported python type - we only support '
-                             f'primitive types at this time. {err}') from err
+                            f'primitive types at this time. {err}') from err
     return params
+
+def get_function_source_definition(func: Callable) -> str:
+    """Needed"""
```
Comment (allegra-noto) [inline AutoMLOps/BuilderUtils.py:259]: needs docstring

File: AutoMLOps/ComponentBuilder.py [hunk 2]
```diff
@@ -115,85 +125,105 @@ def create_component(component_spec: dict,
...
+    f'''    python3 -m ensurepip || python3 -m ensurepip --user || apt-get install python3-pip{newline}'''
+    f'''fi{newline}'''
+    f'''PIP_DISABLE_PIP_VERSION_CHECK=1 python3 -m pip install --quiet \{newline}'''
+    f'''    --no-warn-script-location {concat_package_list} && "$0" "$@"{newline}'''
+    f'''{newline}''')
+    src_code = BuilderUtils.get_function_source_definition(func)
+    return ['sh', '-c', install_python_packages_script, src_code]
+
+def _get_function_parameters(func: Callable) -> dict:
+    """Needed"""
```
Comment (allegra-noto) [inline AutoMLOps/ComponentBuilder.py:193]: needs docstring

File: AutoMLOps/BuilderUtils.py (hunk 1 above)
Comment (srastatter) [reply AutoMLOps/BuilderUtils.py]: Done

File: AutoMLOps/ComponentBuilder.py (hunk 2 above)
Comment (srastatter) [reply AutoMLOps/ComponentBuilder.py]: Done

**Pull request merged**